python/cugraph/structure/graph.py

```diff
@@ -37,32 +58,25 @@ def __init__(self):
         >>> import cuGraph
         >>> G = cuGraph.Graph()
         """
-        self.graph_ptr = graph_wrapper.allocate_cpp_graph()
-
-        self.edge_list_source_col = None
-        self.edge_list_dest_col = None
-        self.edge_list_value_col = None
-
-        self.adj_list_offset_col = None
-        self.adj_list_index_col = None
-        self.adj_list_value_col = None
-
-    def __del__(self):
-        self.delete_edge_list()
-        self.delete_adj_list()
-        self.delete_transposed_adj_list()
-
-        graph_wrapper.release_cpp_graph(self.graph_ptr)
+        self.symmetrized = symmetrized
+        self.renumbered = renumbered
+        self.bipartite = bipartite
+        self.multi = multi
+        self.dynamic = dynamic
+        self.edgelist = None
+        self.adjlist = None
+        self.transposedadjlist = None
+        #self.number_of_vertices = None
 
     def clear(self):
         """
         Empty this graph. This function is added for NetworkX compatibility.
         """
-        self.delete_edge_list()
-        self.delete_adj_list()
-        self.delete_transposed_adj_list()
+        self.edgelist = None
+        self.adjlist = None
+        self.transposedadjlist = None
 
-    def add_edge_list(self, source_col, dest_col, value_col=None, copy=False):
+    def add_edge_list(self, input_df):
```

Comment:
We need something closer to this : https://networkx.github.io/documentation/stable/reference/generated/networkx.convert_matrix.from_pandas_edgelist.html#networkx.convert_matrix.from_pandas_edgelist

In addition, we should accept a renumbering parameter (`True` by default to call renumbering before storing the edge list) and `property` info (to set the flag seen at 61-69).

Reply:
i thought we were following https://networkx.github.io/documentation/stable/reference/classes/generated/networkx.Graph.add_edges_from.html#networkx.Graph.add_edges_from 
with ebunch_to_add being our dataframe

Reply:
The `from_panda_edgelist` style looks more suited for cudf dataframe and cuGraph's main workflow. In case the users have a DF with 10 columns they can still pass it to cuGraph by telling what columns are relevant.

Reply:
fixed. Renumber is set to False by default. User needs to set it to True for renumbering.